### PR TITLE
Fix order var typo in invoice mailer and add basic spec for coverage.

### LINKED ISSUE #3342: Email injection bug from #3303
PR #3303 referenced a non-existent variable. https://github.com/LocalOrbit/localorbit/pull/3303/files#r205301309

### PR DESCRIPTION
**Pull request by rbarreca**
Fixes #3342.

### CODE REVIEW

File: spec/mailers/order_mailer_spec.rb [hunk 1]
```diff
@@ -255,4 +255,14 @@
       end
     end
   end
+
+  describe '.invoice' do
+    before do
+      @notification = OrderMailer.invoice(order.reload.id)
+    end
```
Comment (thermistor) [inline spec/mailers/order_mailer_spec.rb]: why reload?

Comment (thermistor) [reply spec/mailers/order_mailer_spec.rb]: Also we should not use instance variables in before blocks, it's [better](http://www.betterspecs.org/#let) to use `let`:
```
describe '.invoice' do
  let(:notification) { OrderMailer.invoice(order.id) }
  
  it 'delivers to all users in the buyer organization' do
    expect(notification).to deliver_to(buyer_user.email)
  end
end

Comment (rbarreca) [reply spec/mailers/order_mailer_spec.rb]: Good call. I was on copy-past auto-pilot from the gnarly specs above it. Will clean up.

File: app/mailers/order_mailer.rb [hunk 2]
```diff
@@ -56,7 +56,7 @@ def invoice(order_id)
 
     attachments["invoice.pdf"] = {mime_type: "application/pdf", content: @order.invoice_pdf.try(:data)}
 
-    to_list = recipient_list(order)
+    to_list = recipient_list(@order)
```
Comment (ivar) [inline app/mailers/order_mailer.rb]: sorry 😞

**Pull request merged**